go/buildhelper/node.go

```diff
@@ -0,0 +1,207 @@
+package buildhelper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/helpertypes"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+var (
+	nonceLock sync.RWMutex
+)
+
+type EthNode struct {
+	port      uint
+	ipaddress string
+	apiClient *EthAPI
+	id        common.Address
+}
+
+func NewEthNode(id common.Address, ipaddress string, port uint) (obscurocommon.L1Node, error) {
+	apiClient := NewEthAPI(ipaddress, port)
+	err := apiClient.Connect()
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to the eth node - %w", err)
+	}
+
+	log.Log(fmt.Sprintf("Initializing eth node at contract: %s", buildconstants.CONTRACT_ADDRESS))
+	return &EthNode{
+		ipaddress: ipaddress,
+		port:      port,
+		apiClient: apiClient,
+		id:        id,
+	}, nil
+}
+
+func (e *EthNode) RPCBlockchainFeed() []*types.Block {
```

Comment:
@otherview This isn't a feed, it's a snapshot. Should we change the name?

Reply:
I don't mind, but maybe it can change when the way it works gets refactored ? 
This approach might not be viable for feeding millions of blocks into the L1 .

Reply:
there's a task to make it a feed. 
I named it like that initially to not forget that this has to become a stream.